src/main.rs

```diff
@@ -52,14 +54,28 @@ fn exec(exe: OsString, args: Vec<OsString>, argv_skip: usize) -> ExitResult {
         .map(|_| ())
 }
 
-fn main() -> ExitResult {
-    env_logger::init();
-
-    let current_exe = current_exe().map_err(|e| {
+fn find_current_exe() -> Result<PathBuf, Exit> {
+    if let Some(arg) = env::args().next() {
+        let argv0 = PathBuf::from(arg);
+        if argv0.is_file() {
```

Comment:
This is what gets symlinks working. The `std::env::current_exe` used below as the fallback is always fully resolved.